api/services/ocean/spark/schemas/controllerClusterId.yaml

```diff
@@ -0,0 +1,4 @@
+type: string
+example: "my-ocean-cluster"
+description: >
+  The reporting identifier of the Ocean controller in the Ocean cluster
```

Comment:
As a user, I don't understand the differences between the Ocean controller ID and the Ocean ID.

Can we hide one of the IDs or do users need both?
If they need both, it would be nice to add a bit of description. 
For example, are both IDs immutable? Are they both associated for the lifespan of an Ocean cluster? Or at which condition could they change (e.g. can I uninstall and reinstall a controller, and therefore keep the ocean cluster Id but have my controller ID updated)?

Reply:
The Ocean controller ID is stable and can be set by the user.
The Ocean ID is generated by the system.

@thorsteinnth can you let us know when the user needs to know each of them? I think the Ocean controller ID is the most visible as it's the name of the cluster in the UI, am I correct?

Also, we thought we could use the controller ID to trash and recreate a cluster while keeping all associated data (Ocean ID changes but Ocean controller ID stays the same), but there is an issue with that use case that we need to sort out.

Reply:
* The `controllerClusterId` is called cluster identifier in Ocean and it is the main id used by the user. Example from my cluster: `julien-dev`. It cannot be edited. Note that there is a cluster name that I believe can be edited.
* The `oceanClusterId` is system-generated and used by the user when requesting the API (in the path). Example from my cluster: `o-bd661505`

So both IDs are useful in general. In Ocean Spark however, I believe the `oceanClusterId` is not used. @thorsteinnth do you confirm? In which case, could we remove it from the response?

Reply:
Sorry for responding late.

- The `controllerClusterID` is chosen by the user. This is an ID that is known to the cluster (saved on the cluster), and all data sent to the backend is tagged with this ID. By default, this ID is derived from the Ocean cluster name (replacing spaces with hyphens etc.). If you delete your cluster, you can re-use this ID for the next cluster.
- The `oceanClusterID` is system generated, of the form `o-something`. This is the ID of the Ocean cluster entity in Spot's backend. If you delete your cluster and create a new one, you will get a new `oceanClusterID` (while the  `controllerClusterID` could be the same). The user uses this ID when interacting with the Ocean API.

The way I think about it is that the `controllerClusterID` is the ID of the actual k8s cluster, not tied to a specific Spot product. Then we have the Ocean cluster ID which represents the Ocean cluster entity, and the Ocean for Apache Spark cluster ID which represents the OfAS cluster entity.

Reply:
> The controllerClusterId is called cluster identifier in Ocean
The Ocean API also uses the term `controllerClusterId`: https://docs.spot.io/api/#operation/OceanAWSClusterGet

Reply:
The `oceanClusterId` is indeed not strictly necessary for the OfAS user. However, it is used by the UI, in order to link to the corresponding Ocean cluster ("view cluster in Ocean" on the cluster details page).

Reply:
Good catch, thanks! So we'll have to split endpoints or add an internal flag before we can do remove the `oceanClusterId`.
So let's keep it for now.